packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -401,6 +422,43 @@ export class WorkspaceCommandContribution implements CommandContribution {
         return false;
     }
 
+    /**
+     * Save source `URI` to target.
+     *
+     * @param uri the source `URI`.
+     */
+    protected async saveAs(uri: URI): Promise<void> {
+        let exist: boolean = false;
+        let overwrite: boolean = false;
+        let target: URI | undefined;
+        const stat = await this.fileSystem.getFileStat(uri.toString(true));
+        do {
+            target = await this.fileDialogService.showSaveDialog(
+                {
+                    title: 'Save As',
+                    filters: {},
+                    inputValue: uri.path.base
+                }, stat);
+            if (target) {
+                exist = await this.fileSystem.exists(target.toString());
+                if (exist) {
+                    overwrite = await this.confirmOverwrite(target);
+                }
+            }
+        } while (target && exist && !overwrite);
+        if (target) {
+            this.fileSystem.copy(uri.toString(), target.toString(), { overwrite: overwrite });
+        }
+    }
+
+    private async confirmOverwrite(uri: URI): Promise<boolean> {
```

Comment:
How about implementing it in `WorkspaceFrontendContribution` then you don't need to copy code and create new keybinding contributions.